### DISCUSSION TITLE
Dealing with Writes from Clients

### DISCUSSION THREAD

**Discussion by jamestjsp** [FreeOpcUa/opcua-asyncio#1580] [Q&A]
I am developing an OPC UA server for an online application. The application has API for get and put data. I am wondering how put data to the API if an OPC UA client writes to the Python OPC UA server?

The use case is similar to what described in the link below.
https://opcfoundation.github.io/UA-.NETStandard/help/step_4_-_handling_write_requests.htm

**Comment by oroulet** [comment]
Not sure what you want exactly but subscribing to the data server side is usually what I would do

**Reply by jamestjsp** [reply]
When client writes values to an OPC variable I need to detect the event in server and put the value to my soft controller. Just like the PLC use case.

**Reply by oroulet** [reply]
Yes then use subscription on server side look at the examples. Same API server and client side

**Reply by jamestjsp** [reply]
Thanks this works. There is struggle to identify whether a value is changed by client or Server :)
